mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -672,7 +672,10 @@ struct WMMAOpLowering : public ConvertOpToLLVMPattern<WMMAOp> {
                   ConversionPatternRewriter &rewriter) const override {
     Location loc = op.getLoc();
     auto outType =
-        cast<VectorType>(typeConverter->convertType(op.getDestD().getType()));
+        typeConverter->convertType<VectorType>(op.getDestD().getType());
+    if (!outType)
+      return rewriter.notifyMatchFailure(
+          op, "wmma output doesn't convert to a vector for no clear reason");
```

Comment:
or just
```suggestion
          op, "type conversion failed");
```
(this is what we do in spirv lowering)